### PR TITLE
[FLINK-11633][docs-zh] Translate "Working with state" into Chinese

### PR DESCRIPTION
**Pull request by klion26**
## What is the purpose of the change

Translate "Working with state" into Chinese

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by klion26** [comment]
This page contains many pieces of information, so I'll recheck the translation again, and any reviews are welcome.
@wuchong

File: docs/dev/stream/state/state.zh.md [hunk 1]
```diff
@@ -22,122 +22,87 @@ specific language governing permissions and limitations
...
+*Keyed State* 通常和键相关，仅可使用在 `KeyedStream` 的方法和算子中。
 
-You can think of Keyed State as Operator State that has been partitioned,
-or sharded, with exactly one state-partition per key.
-Each keyed-state is logically bound to a unique
-composite of <parallel-operator-instance, key>, and since each key
-"belongs" to exactly one parallel instance of a keyed operator, we can
-think of this simply as <operator, key>.
+你可以把 Keyed State 看作分区或者共享的 Operator State, 而且每个键仅出现在一个分区内。
+逻辑上每个 keyed-state 和 <算子并发实例, key> 相绑定，由于每个 key 仅"属于"
```
Comment (klion26) [inline docs/dev/stream/state/state.zh.md]: maybe `逻辑上每个 keyed-state 和 <算子并发实例, key> 相绑定` can be `逻辑上每个 keyed-state 和 <算子并发实例, key> 绑定`

`由于每个 key 仅"属于"` --> `另外每个 key 仅"属于"`

File: docs/dev/stream/state/state.zh.md [hunk 2]
```diff
@@ -22,122 +22,87 @@ specific language governing permissions and limitations
...
+算子的一个并发，因此简化为 <算子, key>。
 
-Keyed State is further organized into so-called *Key Groups*. Key Groups are the
-atomic unit by which Flink can redistribute Keyed State;
-there are exactly as many Key Groups as the defined maximum parallelism.
-During execution each parallel instance of a keyed operator works with the keys
-for one or more Key Groups.
+Keyed State 会按照 *Key Group* 进行管理。Key Group 是 Flink 分发 Keyed State 的最小单元；
+Key Groups 的数目等于作业的最大并发数。在执行过程中，每个 keyed operator 会对应到一个或多个 Key Group
```
Comment (klion26) [inline docs/dev/stream/state/state.zh.md]: `Key Groups` can be `Key Group`?

File: docs/dev/stream/state/state.zh.md [hunk 3]
```diff
@@ -22,122 +22,87 @@ specific language governing permissions and limitations
...
 
 ### Operator State
 
-With *Operator State* (or *non-keyed state*), each operator state is
-bound to one parallel operator instance.
-The [Kafka Connector]({{ site.baseurl }}/dev/connectors/kafka.html) is a good motivating example for the use of Operator State
-in Flink. Each parallel instance of the Kafka consumer maintains a map
-of topic partitions and offsets as its Operator State.
+对于 *Operator State* (或者 *non-keyed state*) 来说，每个 operator state 和一个并发实例进行绑定。
+[Kafka Connector]({{ site.baseurl }}/zh/dev/connectors/kafka.html) 是 Flink 中使用 operator state 的一个很好的示例。
```
Comment (klion26) [inline docs/dev/stream/state/state.zh.md]: `是 Flink 中使用 operator state 的一个很好的示例。` --> `是 Flink 中一个使用 operator state 的好示例。`

File: docs/dev/stream/state/state.zh.md [hunk 4]
```diff
@@ -22,122 +22,87 @@ specific language governing permissions and limitations
...
+每个 Kafka 消费者的并发在 Operator State 中维护一个 topic partition 到 offset 的映射关系。
 
-The Operator State interfaces support redistributing state among
-parallel operator instances when the parallelism is changed. There can be different schemes for doing this redistribution.
+Operator State 在 Flink 作业的并发改变后，会重新分发状态，分发的策略和 Keyed State 不一样。
```
Comment (klion26) [inline docs/dev/stream/state/state.zh.md]: `Operator State 在 Flink 作业的并发改变后` --> `Operator State 在 Flink 作业改变并发后`

File: docs/dev/stream/state/state.zh.md [hunk 5]
```diff
@@ -22,122 +22,87 @@ specific language governing permissions and limitations
...
 
 ## Raw and Managed State
 
-*Keyed State* and *Operator State* exist in two forms: *managed* and *raw*.
+*Keyed State* 和 *Operator State* 分别有两种存在形式：*managed* and *raw*.
 
-*Managed State* is represented in data structures controlled by the Flink runtime, such as internal hash tables, or RocksDB.
-Examples are "ValueState", "ListState", etc. Flink's runtime encodes
-the states and writes them into the checkpoints.
+*Managed State* 有 Flink runtime 中的数据结构所控制，比如内部的 hash table 或者 RocksDB。
```
Comment (klion26) [inline docs/dev/stream/state/state.zh.md]: `有 Flink runtime ` --> `由 Flink runtime `

File: docs/dev/stream/state/state.zh.md [hunk 6]
```diff
@@ -22,122 +22,87 @@ specific language governing permissions and limitations
...
-*Raw State* is state that operators keep in their own data structures. When checkpointed, they only write a sequence of bytes into
-the checkpoint. Flink knows nothing about the state's data structures and sees only the raw bytes.
+*Raw State* 则保存在算子自己的数据结构中。checkpoint 的时候，Flink 并不知晓具体的内容，仅仅写入一串字节序列到 checkpoint。
 
-All datastream functions can use managed state, but the raw state interfaces can only be used when implementing operators.
-Using managed state (rather than raw state) is recommended, since with
-managed state Flink is able to automatically redistribute state when the parallelism is
-changed, and also do better memory management.
+所有 datastream 的方法都可以使用 managed state, 但是 raw state 则只能在实现算子的时候使用。
+由于 Flink 可以在修改并发是更好的重新分发状态数据，并且能够更好的管理内存，因此建议使用 managed state（而不是 raw state）。
```
Comment (klion26) [inline docs/dev/stream/state/state.zh.md]: `由于 Flink 可以在修改并发是更好的重新分发状态数据` --> `由于 Flink 可以在修改并发时更好的分发状态数据`

File: docs/dev/stream/state/state.zh.md [hunk 7]
```diff
@@ -22,122 +22,87 @@ specific language governing permissions and limitations
...
 
-<span class="label label-danger">Attention</span> If your managed state needs custom serialization logic, please see 
-the [corresponding guide](custom_serialization.html) in order to ensure future compatibility. Flink's default serializers 
-don't need special treatment.
+<span class="label label-danger">注意</span> 如果你的 managed state 需要定制化的序列化逻辑，
+为了后续的兼容性请参考 [corresponding guide](custom_serialization.html)，Flink 默认提供的序列化器不需要用户做特殊的处理。
```
Comment (klion26) [inline docs/dev/stream/state/state.zh.md]: `corresponding guide` --> `相应指南`

`Flink 默认提供的序列化器` --> `Flink 的默认序列化器`

File: docs/dev/stream/state/state.zh.md [hunk 8]
```diff
@@ -22,122 +22,87 @@ specific language governing permissions and limitations
...
 
 ## Using Managed Keyed State
 
-The managed keyed state interface provides access to different types of state that are all scoped to
-the key of the current input element. This means that this type of state can only be used
-on a `KeyedStream`, which can be created via `stream.keyBy(…)`.
+managed keyed state 接口提供不同类型状态访问的接口，这些状态都限定于当前的输入数据。换句话说，这些状态仅可在 `KeyedStream`
```
Comment (klion26) [inline docs/dev/stream/state/state.zh.md]: `不同类型状态访问的接口` --> `不同类型状态的访问接口`

`这些状态都限定于当前的输入数据` --> `这些状态以当前输入数据为键`

File: docs/dev/stream/state/state.zh.md [hunk 9]
```diff
@@ -394,30 +338,25 @@ val ttlConfig = StateTtlConfig
...
+该特性可以通过 `StateTtlConfig` 进行启用：
 
 <div class="codetabs" markdown="1">
 <div data-lang="java" markdown="1">
 {% highlight java %}
 import org.apache.flink.api.common.state.StateTtlConfig;
  StateTtlConfig ttlConfig = StateTtlConfig
     .newBuilder(Time.seconds(1))
-    .cleanupIncrementally()
+    .cleanupIncrementally(10, true)
```
Comment (wuchong) [inline docs/dev/stream/state/state.zh.md]: Please also update scala example and `state.md`.

File: docs/dev/stream/state/state.zh.md [hunk 10]
```diff
@@ -524,46 +443,32 @@ val counts: DataStream[(String, Int)] = stream
...
-only the place where different types of state are initialized, but also where state recovery logic is included.
+进行 checkpoint 时会调用 `snapshotState()`。 用户自定义函数初始化时会调用 `initializeState()`，初始化包括第一次自定义函数初始化和从之前的 checkpoint 恢复。
+因此 `initializeState()` 不仅是定义不同状态类型初始化的地方，也需要包括状态恢复的逻辑。
 
-Currently, list-style managed operator state is supported. The state
-is expected to be a `List` of *serializable* objects, independent from each other,
-thus eligible for redistribution upon rescaling. In other words, these objects are the finest granularity at which
-non-keyed state can be redistributed. Depending on the state accessing method,
-the following redistribution schemes are defined:
+当前，managed operator state 已 list 的形式存在。这些状态是一个 *可序列* 对象的集合，彼此独立，方便在改变并发后进行状态的重新分派。
```
Comment (wuchong) [inline docs/dev/stream/state/state.zh.md]:
```suggestion
当前，managed operator state 以 list 的形式存在。这些状态是一个 *可序列化* 对象的集合 `List`，彼此独立，方便在改变并发后进行状态的重新分派。
```

File: docs/dev/stream/state/state.zh.md [hunk 11]
```diff
@@ -703,43 +605,33 @@ checkpointedState = context.getOperatorStateStore.getListState(descriptor)
...
-recovering after a failure. If this is `true`, *i.e.* we are recovering, the restore logic is applied.
+调用不同的获取状态对象的接口，会使用不同的状态分配算法。比如 `getUnionListState(descriptor)` 会使用 union redistribution 算法，
+而 `getListState(descriptor)` 则简单的使用 even-split redistribution 算法。
+
+当初始化好状态对象后，我们通过 `isRestored()` 方法判断是否从之前的故障中恢复回来，如果该方法返回 `true` 则表示从故障中进行恢复，会执行接下来的恢复逻辑。
 
-As shown in the code of the modified `BufferingSink`, this `ListState` recovered during state
-initialization is kept in a class variable for future use in `snapshotState()`. There the `ListState` is cleared
-of all objects included by the previous checkpoint, and is then filled with the new ones we want to checkpoint.
+正入代码所示，`BufferingSink` 中初始化时，恢复回来的 `ListState` 的所有元素会添加到一个局部变量中，供下次 `snapshotState()` 时使用。
```
Comment (wuchong) [inline docs/dev/stream/state/state.zh.md]:
```suggestion
正如代码所示，`BufferingSink` 中初始化时，恢复回来的 `ListState` 的所有元素会添加到一个局部变量中，供下次 `snapshotState()` 时使用。
```

File: docs/dev/stream/state/state.zh.md [hunk 12]
```diff
@@ -703,43 +605,33 @@ checkpointedState = context.getOperatorStateStore.getListState(descriptor)
...
 {% endhighlight %}
 
-On `snapshotState()` the operator should return a list of objects to checkpoint and
-`restoreState` has to handle such a list upon recovery. If the state is not re-partitionable, you can always
-return a `Collections.singletonList(MY_STATE)` in the `snapshotState()`.
+`snapshotState()` 需要返回一个将写入到 checkpoint 的对象列表，`restoreState` 则需要处理恢复回来的对象列表。如果状态不可切分，
+则可以在 `snapshotState()` 中返回 `Collections.singletonList(MY_STATE)`。
 
-### Stateful Source Functions
+### 带状态的输入函数
```
Comment (wuchong) [inline docs/dev/stream/state/state.zh.md]:
```suggestion
### 带状态的 Source Function
```
Keep Source Function untranslated?

**Comment by klion26** [comment]
@wuchong thanks for the review, I've updated according to the comments.
I think we can let `managed state` and `raw state` as it is, seems like `keyed state` and `operator state`

**Comment by wuchong** [comment]
Merged.

**Comment by klion26** [comment]
@wuchong thanks for the review and advice.

**Pull request closed without merging**